node/core/av-store/src/lib.rs

```diff
@@ -0,0 +1,516 @@
+						Ok(FromOverseer::Communication { msg }) => {
+							let _ = self.process_message(msg);
+						}
+						Err(_) => break,
+					}
+				}
+				complete => break,
+			}
+		}
+
+		Ok(())
+	}
+
+	fn process_message(&mut self, msg: AvailabilityStoreMessage) -> Result<(), Error> {
+		use AvailabilityStoreMessage::*;
+		match msg {
+			QueryAvailableData(hash, tx) => {
+				let _ = tx.send(self.available_data(&hash).map(|d| d.data));
+			}
+			QueryChunk(hash, id, tx) => {
+				let _ = tx.send(self.get_chunk(&hash, id)?);
+			}
+			StoreChunk(hash, id, chunk) => {
+				self.store_chunk(&hash, id, chunk)?;
+			}
+			StoreAvailableData(hash, id, n_validators, av_data) => {
+				self.store_available_data(&hash, id, n_validators, av_data)?;
+			}
+		}
+
+		Ok(())
+	}
+
+	fn available_data(&self, candidate_hash: &Hash) -> Option<StoredAvailableData> {
+		self.query_inner(columns::DATA, &available_data_key(candidate_hash))
+	}
+
+	fn store_available_data(
+		&self,
+		candidate_hash: &Hash,
+		id: Option<ValidatorIndex>,
+		n_validators: u32,
+		available_data: AvailableData,
+	) -> Result<(), Error> {
+		let mut tx = DBTransaction::new();
+
+		if let Some(index) = id {
+			let chunks = get_chunks(&available_data, n_validators as usize)?;
+			self.store_chunk(candidate_hash, n_validators, chunks[index as usize].clone())?;
+		}
+
+		let stored_data = StoredAvailableData {
+			data: available_data,
+			n_validators,
+		};
+
+		tx.put_vec(
+			columns::DATA,
+			available_data_key(&candidate_hash).as_slice(),
+			stored_data.encode(),
+		);
+
+		self.inner.write(tx)?;
+
+		Ok(())
+	}
+
+	fn store_chunk(&self, candidate_hash: &Hash, _n_validators: u32, chunk: ErasureChunk)
+		-> Result<(), Error>
+	{
+		let mut tx = DBTransaction::new();
+
+		let dbkey = erasure_chunk_key(candidate_hash, chunk.index);
+
+		tx.put_vec(columns::DATA, &dbkey, chunk.encode());
+		self.inner.write(tx)?;
+
+		Ok(())
+	}
+
+	fn get_chunk(&self, candidate_hash: &Hash, index: u32) -> Result<Option<ErasureChunk>, Error> {
+		if let Some(chunk) = self.query_inner(
+			columns::DATA,
+			&erasure_chunk_key(candidate_hash, index)) {
+			return Ok(Some(chunk));
+		}
+
+		if let Some(data) = self.available_data(candidate_hash) {
+			let mut chunks = get_chunks(&data.data, data.n_validators as usize)?;
+			let chunk = chunks.get(index as usize).cloned();
+			for chunk in chunks.drain(..) {
+				self.store_chunk(candidate_hash, data.n_validators, chunk)?;
+			}
+			return Ok(chunk);
+		}
+
+		Ok(None)
+	}
+
+	fn query_inner<D: Decode>(&self, column: u32, key: &[u8]) -> Option<D> {
+		match self.inner.get(column, key) {
+			Ok(Some(raw)) => {
+				let res = D::decode(&mut &raw[..]).expect("all stored data serialized correctly; qed");
+				Some(res)
+			}
+			Ok(None) => None,
+			Err(e) => {
+				log::warn!(target: LOG_TARGET, "Error reading from the availability store: {:?}", e);
+				None
+			}
+		}
+	}
+}
+
+/// Configuration for the availability store.
+pub struct Config {
```

Comment:
Is this `Config` used? A helper constructor for the `AvailabilityStoreSubsystem` that uses it to create one that uses rocksdb under the hood could be useful.

Reply:
Yes, there is a `pub fn new(config: Config) -> io::Result<Self>`